### DISCUSSION TITLE
TypeError: Cannot read property 'reference' of undefined     at EntityLoader.populateMany

### DISCUSSION THREAD

**Discussion by n4an** [mikro-orm/mikro-orm#2520] [Q&A] [answered]
TypeError: Cannot read property 'reference' of undefined
    at EntityLoader.populateMany (/var/www/html/selbi.angular.nest/server/node_modules/@mikro-orm/core/entity/EntityLoader.js:108:18)

**Comment by n4an** [comment, marked as answer]
When to use Collection and IdentifiedReference?

**Comment by B4nan** [comment]
That error almost always means you have something wrong in the metadata, hard to say what as you didnt share the full stack trace, so I have no idea where it was thrown. 

> When to use Collection

Always for collections, so for 1:m and m:n relations, it is required for those.

> When to use IdentifiedReference?

If you want added type safety, wrapped references are optional - and they work only for m:1 and 1:1 relations. Note you need to have `wrappedReference: true` in the decorator to use it (or use ts-morph which can infer it).

> how to populate nested and not extract all fields
> {populate: { roles: { fileds: ['id', name], where: {isActive: true }, includes: { permissions: { fields: ['name'], where: { isActive: true } } } }

Not sure what that is, there is no such thing as `includes` in the ORM codebase. Isn't that from some other ORM? Also you can't nest things this way, `fields` is a top level option. This code should not even type check, at least if you have strict codebase.
